WooCommerce/Classes/Authentication/AuthenticatedWebViewController.swift

```diff
@@ -157,42 +178,98 @@ private extension AuthenticatedWebViewController {
 
         webView.publisher(for: \.url)
             .sink { [weak self] url in
-                guard let self else { return }
-                let initialURL = self.viewModel.initialURL
-                // avoids infinite loop if the initial url happens to be the nonce retrieval path.
-                if url?.absoluteString.contains(WKWebView.wporgNoncePath) == true,
-                   initialURL?.absoluteString.contains(WKWebView.wporgNoncePath) != true {
-                    self.loadContent()
-                } else {
-                    self.viewModel.handleRedirect(for: url)
-                }
+                guard let url else { return }
+                self?.handleRedirect(for: url)
             }
             .store(in: &subscriptions)
+    }
 
-        if let siteCredentials, let request = try? webView.authenticateForWPOrg(with: siteCredentials) {
-            webView.load(request)
-        } else {
-            loadContent()
+    /// Authentication logic differs depending on the destination URL and the current site.
+    /// More information: https://wp.me/pe5sF9-3Si
```

Comment:
```suggestion
    /// More information: pe5sF9-3Si-p2
```

Reply:
Nice catch! Fixed in e9d83dfd873b371c58a91bee1c30592d9741578a.